api/v1alpha1/serviceintentions_types_test.go

```diff
@@ -641,10 +643,20 @@ func TestServiceIntentions_Default(t *testing.T) {
 		},
 	}
 	for name, testCase := range cases {
-		t.Run(name, func(t *testing.T) {
-			testCase.input.Default()
-			require.True(t, cmp.Equal(testCase.input, testCase.output))
-		})
+		// We also test with consul namespaces enabled/disabled. When disabled,
+		// the input shouldn't change since we don't set any namespace defaults
+		// in OSS.
+		for _, namespacesEnabled := range []bool{false, true} {
+			testName := fmt.Sprintf("%s namespaces=%t", name, namespacesEnabled)
+			t.Run(testName, func(t *testing.T) {
+				testCase.input.Default(namespacesEnabled)
+				if namespacesEnabled {
+					require.True(t, cmp.Equal(testCase.input, testCase.output))
+				} else {
+					require.True(t, cmp.Equal(testCase.input, testCase.input))
```

Comment:
I think this assertion will always be true.

Reply:
Isn't that the best kind of assertion?

Reply:
i mean if you phrase it that way 😂